pandas/tests/indexing/test_loc.py

```diff
@@ -752,3 +753,21 @@ def convert_nested_indexer(indexer_type, keys):
             index=pd.MultiIndex.from_product(keys))
 
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        'indexer,value', itertools.product(
+            [['A'], slice(None, 'A', None), np.array(['A'])],
+            [['Z'], np.array(['Z'])]))
+    def test_loc_setitem_with_scalar_index(self, indexer, value):
+        # GH #19474
+        # assigning like "df.loc[0, ['A']] = ['Z']" should be evaluated
+        # elementwisely, not using "setter('A', ['Z'])".
+
+        df = pd.DataFrame([[1, 2], [3, 4]], columns=['A', 'B'])
+        df.loc[0, indexer] = value
+        result = df.loc[0, 'A']
+
+        if not is_scalar(result) and result == 'Z':
```

Comment:
I think it'd be more direct to just do `assert result == 'Z'`.

Usually in tests you just assert what you want the result to be, instead of catching what you don't want and writing an informative error message like you would in user facing code.  If the test does fail, `pytest` will automatically give details similar to the message you wrote.

Reply:
Thank you for comments @jschendel 

I rewrote codes,  but kept  `is_scalar(result)`. Without it, If `result` is `np.array(['Z'])` , we cannot catch error because of broadcasting.

BTW, I should have written the conditional statement as `not is_scalar(result) and not result == 'Z'`. Sorry.